recipe-server/normandy/recipes/api/v1/serializers.py

```diff
@@ -172,6 +172,10 @@ def validate_extra_filter_expression(self, value):
         return value
 
     def validate_arguments(self, value):
+        # Ensure the value is a dict
+        if not isinstance(value, dict):
+            raise serializers.ValidationError('Invalid arguments.')
```

Comment:
We can probably have a better error message than this. Maybe "Must be an object"? Bonus points if you use DRF's [`self.fail`](http://www.django-rest-framework.org/api-guide/fields/#raising-validation-errors).

Reply:
I opted not to go with `self.fail` because it actually seemed like not a great way of doing this given that you can't just pass a message in and have to rely on default error messages.